integreat_cms/api/v3/events.py

```diff
@@ -185,16 +188,20 @@ def events(request, region_slug, language_slug):
     region = request.region
     # Throw a 404 error when the language does not exist or is disabled
     region.get_language_or_404(language_slug, only_active=True)
+
     result = []
     now = timezone.now().date()
     for event in region.events.prefetch_public_translations().filter(archived=False):
         event_translation = event.get_public_translation(language_slug)
         if event_translation:
-            if event.end_local.date() >= now:
+            if event.end_local.date() >= now or (
+                event.is_recurring
+                and event.recurrence_rule.to_ical_rrule().after(timezone.now(), True)
+            ):  # or is recurring and last recurring is >= now
```

Comment:
I added this comment, but in general i don't think these types of comments are helpful or are any good when thinking about clean code principles.

Reply:
Ok, for me it would also be fine to remove the comment.
But the previous version with the hanging inline comment after a multi-line if-statement looked less readable to me, this is why I suggested to move the comment into its own line.
Maybe it would be even better to create a new property to the event model which explains what this if-condition is doing? E.g. something along the lines of:

```
@cached_property
def is_past(self):
    """
    This property checks whether an event lies in the past, including potential future recurrences.

    :return: Whether event lies in the past
    :rtype: bool
    """
    now = timezone.now()
    duration = event.end_local - event.start_local
    future_recurrence = (
        self.is_recurring
        and self.recurrence_rule.to_ical_rrule().after(now - duration, True)
    )
    return self.end_local.date() < now.date() and not future_recurrence
```

then, we could use the simpler and more readable property in the API:

```
if not event.is_past:
    result.append(transform_event_translation(event_translation))
```

I also noticed that the `rrule.after()` method only takes the start datetime into account, so in case the last occurrence is currently ongoing, the condition `after(now)` would not return the current instance. So I suggest to calculate the potential start date of the currently ongoing occurrence by subtracting the duration of the event from the current time.
This still has the caveat of an inconsistency (non-recurring events are still shown on their last day even after they're over, and recurring events disappear as soon as the end time has been reached), but I'm not sure which of both behaviors is the desired one?